main.cpp

```diff
@@ -125,372 +170,568 @@ struct CarWash
+    }
+}
+
+void House::openFrontDoor(std::string door)
+{
+    if ( door == "front door" )
+    {
+        std::cout << "Need to open front door" << std::endl;
+    }
+
+    else
+    {
+        std::cout << "Not the front door" << std::endl;
+    }
+}
+
+bool House::closeWindow(House::Kitchen myKitchen)
+{
+    if ( myKitchen.numShelves > 1 )
+    {
+        return true; 
+    }
+
+    else
+    {
+        return false;
+    }
+
+}
 
 struct Xbox
 {
-    // amount of space (float)
     float amountOfSpace = 500.f;
-    // number of games downloaded (int)
     int numGamesDownloaded = 10;
-    // number of controllers (int)
     int numControllers = 1;
-    // name of profile (std::string)
     std::string profileName = "Profile1";
-    // number of friends online (int)
     int numFriendsOnline = 8;
 
     struct Game
     {
-        // size of game (float)
         float sizeOfGame = 55.5f;
-        // category of game (std::string)
         std::string gameCategory = "RPG";
-        // rating of game (char)
         char gameRating = 'M';
-        // hours of game played (float)
         float timePlayed = 15.2f;
-        // number of achievements completed (int)
         int numCompletedAchievements = 5;
 
         bool openGame(std::string gameName);
         bool completeAchievement(std::string achievementName, int numPoints = 5);
         void changeMenu(char buttonPressed = 'b', std::string currentMenu = "Main menu");
     };
-    // open game
     bool openGame(Xbox::Game skyrim);
-    // download game
     bool downloadGame(Xbox::Game haloInfinite);
-    // turn on Xbox
     void turnOnXbox(Xbox myXbox);
 };
 
-/*
-Thing 3) Logic Pro
-5 properties:
-    1) project sample rate (float)
-    2) name of input device (std::string)
-    3) name of project (std::string)
-    4) number of tracks (int)
-    5) tempo (double)
-3 things it can do:
-    1) create a new instrument track
-    2) insert plugin
-    3) record performance
- */
+bool Xbox::Game::openGame(std::string gameName)
+{
+    if ( gameName == "Valid" )
+    {
+        return true;
+    }
+
+    else
+    {
+        return false;
+    }
+}
+
+bool Xbox::Game::completeAchievement(std::string achievementName, int numPoints)
+{
+    if ( achievementName == "Valid" )
+    {
+        numPoints += 5;
+        return true;
+    }
+
+    else
+    {
+        numPoints += 0;
+        return false;
+    }
+}
+
+void Xbox::Game::changeMenu(char buttonPressed, std::string currentMenu)
+{
+    if ( buttonPressed == 'b' )
+    {
+        currentMenu = "Main menu";
+    }
+
+}
+
+bool Xbox::openGame(Xbox::Game skyrim)
+{
+    if ( skyrim.gameCategory == "RPG" )
+    {
+        return true;
+    }
+
+    else
+    {
+        return false;
+    }
+}
+
+bool Xbox::downloadGame(Xbox::Game haloInfinite)
+{
+    Xbox myXbox;
+    
+    if ( myXbox.amountOfSpace >= haloInfinite.sizeOfGame )
+    {
+        return true;
+    }
+
+    else
+    {
+        return false;
+    }
+}
```

Comment:
`Xbox::downloadGame` is a member function. 
you do not need to declare another instance of `Xbox` inside of this function to use member variables. 
When you call this function with an instance of `Xbox`, you can use the caller's instance's member variables.
i.e.
```
Xbox xbox;
xbox.downloadGame( Xbox::Game() );
//the 'amountOfSpace' used inside the 'downloadGame' function is the one inside 'xbox'
```
- [ ] revise

Reply:
I don't completely understand what you mean. I made changes to this one but I don't know if I did it correctly.